gateway/ln-gateway/tests/integration_tests.rs

```diff
@@ -53,10 +61,64 @@ async fn gatewayd_supports_multiple_federations() -> anyhow::Result<()> {
 }
 
 #[tokio::test(flavor = "multi_thread")]
-async fn gatewayd_shows_info_about_all_connected_federations() {
-    let (_, _, _, client) = fixtures().await;
-    let info = client.get_info().await.unwrap();
+async fn gatewayd_shows_summary_info_about_all_connected_federations() {
+    let (fed1, fed2, _, client) = fixtures().await;
+    let info = client.get_info(None).await.unwrap();
+
+    let fed_ids: Vec<FederationId> = join_all(
+        vec![fed1, fed2]
+            .iter()
+            .map(|f| async { f.new_client().await.federation_id() }),
+    )
+    .await;
+
+    let info_ids: Vec<FederationId> = info.federations.iter().map(|f| f.federation_id).collect();
+
     assert_eq!(info.federations.len(), 2);
+    assert!(fed_ids.iter().all(|id| info_ids.contains(id)));
+}
+
+#[tokio::test(flavor = "multi_thread")]
+async fn gatewayd_shows_detailed_info_about_filtered_federation() {
+    let (fed1, fed2, _, client) = fixtures().await;
+
+    let fed1_id = fed1.new_client().await.federation_id();
+    let fed1_balance = client
+        .get_balance(BalancePayload {
+            federation_id: fed1_id,
+        })
+        .await
+        .unwrap();
+    let fed1_info = client.get_info(Some(fed1_id)).await.unwrap();
+
+    assert_eq!(fed1_info.federations.len(), 1);
+    assert_eq!(
+        fed1_info.federations.first().unwrap().federation_id,
```

Comment:
nit: tests would read better if you assign id to a variable or expose a function for getting it.